### PR TITLE
Reduce number of places where go version is set

### PR DESCRIPTION
**Pull request by ncabatoff**
Now we have only .go_version and @executors.yml.  Also add a GH action to ensure those two are kept in sync.

Details:
* go_test accepted a go_image parameter but nothing ever specified it, we were just relying on the default, so now instead we'll use the GO_IMAGE environment variable
* reworked @executors to look more like the ent version of the file; ensure that GO_IMAGE environment variable is set everywhere; and set the version once in an anchor, in references.environment.GO_IMAGE
* add a .go-version file in the root of the tree, as some ecosystem repos are doing already, and change the Makefile to use that file instead of hard-coding it; this allows us to programatically determine the current version the repo wants without having to parse the Makefile or invoke make
* remove references to specific Go version from README; we can assume that most people will install the latest Go version, and the Makefile will protect against those who don't
* remove references to specific Go version in dockerfiles; the Makefile target that uses them already passes in the Go version anyway

### CODE REVIEW

**Review by tomhjp — APPROVED** [review]
Some optional suggestions on the action, but this seems like a nice improvement 👍

File: .github/workflows/goversion-checker.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+
+jobs:
+  go-version-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply .github/workflows/goversion-checker.yml:11]: Will update.

File: .github/workflows/goversion-checker.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Check Go version
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply .github/workflows/goversion-checker.yml:5]: I copied and pasted the changelog checker as my starting point.  I can remove them, I agree at the moment there's no point.

File: .github/workflows/goversion-checker.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          fetch-depth: 0
+      - name: Fetch go version used for circleci
+        id: getExecutorImage
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq '.references.environment.GO_IMAGE' '.circleci/config/executors/@executors.yml'
+      - name: Verify go versions in tree are consistent with one another
+        run: |
+          GOVER=$(cat .go-version)
+          EXPECTED="docker.mirror.hashicorp.services/cimg/go:$GOVER"
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply .github/workflows/goversion-checker.yml:23]: I tried to ensure that there's only a single place in circleci config and I somehow missed this one staring me in the face!    I think I can eliminate it in favour of having setup-go.yml read from .go-version.

[the remark this answers is not in the document]
Comment (tomhjp) [reply .github/workflows/goversion-checker.yml:23]: Oh nice, even better! :)

**Pull request merged**